docs/markdown/Freedesktop-module.md

```diff
@@ -0,0 +1,47 @@
+# Freedesktop Module
+
+This module provides helper functions for freedesktop standards used primarily on *nix systems.
+
+*New in 0.55.0*
+
+## File generators
+
+### desktop_file()
+
+```meson
+  datafile desktop_file(exe, ...)
+```
+
+Creates a an datafile that will be installed if the executable used to
+generate it is also installed. Takes one positional argument which is an
+executable or a disabler. If a disabler is passed or the executable is not going to be installed then nothing is returned.
+
+The `Exec` field is automatically populated with the installed path to the executable.
+
+The `Terminal` field is set based on the value of the Executable's `gui_app` field.
+
+This function has the following keyword arguments:
+
+- `name` The name of the program. This is the value of the `Name` field in the .desktop file. If this field is not set then the name of the executable is used.
+
+- `comment` The value of the `Comment` field in the .desktop file. If this value is not provided then the field is unset.
+
+- `generic_name` is the `GenericName` in the .desktop file. If this value is not provided then the field is unset.
+
+- `icon` is the `Icon` value in the .desktop file. Meson does not validate this value in anyway. If not provided the field is unset.
+
+- `categories` a meson list of categories for this program. These values will be joined into a single semicolon delimited string in the .desktop file. Meson does not validate this value in any way. If not provided the field is unset.
+
+- `install_name` The name of the .desktop file without the file extension. If this is unset the name of the value of `name` or the name of the executable will be used, in that order.
```

Comment:
"the name of the value of `name`" sounds weird. Do you mean "the value of the `name` field or the name of the executable..." ?